zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ExecutionListenerJobTest.java

```diff
@@ -0,0 +1,640 @@
+        .ofInstance(processInstanceKey)
+        .withType("endListener1")
+        .withVariable("startEventVar", "aValueUpdated")
+        .complete();
+
+    // then: Validate start event variable updated by the end execution listener
+    assertVariable(
+        processInstanceKey, VariableIntent.UPDATED, "startEventVar", "\"aValueUpdated\"");
+
+    // when: Completing the second end execution listener job
+    ENGINE.job().ofInstance(processInstanceKey).withType("endListener2").complete();
+
+    // then: Validate merged variables after all listeners and tasks are completed
+    assertVariable(
+        processInstanceKey, VariableIntent.CREATED, "mergedVars", "\"aValue+bValueUpdated\"");
+  }
+
+  private static void completeJobByType(
+      final long processInstanceKey, final String jobType, final int jobIndex) {
+    final long jobKey =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withType(jobType)
+            .skip(jobIndex)
+            .getFirst()
+            .getKey();
+    ENGINE.job().ofInstance(processInstanceKey).withKey(jobKey).complete();
+  }
+
+  private void assertVariable(
+      final long processInstanceKey,
+      final VariableIntent intent,
+      final String varName,
+      final String expectedVarValue) {
+    final Record<VariableRecordValue> variableRecordValueRecord =
+        RecordingExporter.variableRecords(intent)
+            .withProcessInstanceKey(processInstanceKey)
+            .withName(varName)
+            .getFirst();
+
+    Assertions.assertThat(variableRecordValueRecord.getValue())
+        .hasName(varName)
+        .hasValue(expectedVarValue);
+  }
+
+  private void assertJobState(
+      final long processInstanceKey,
+      final long jobIndex,
+      final String expectedJobType,
+      final JobIntent expectedJobIntent,
+      final JobKind expectedJobKind) {
+    final Record<ProcessInstanceRecordValue> activatingJob =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+            .withElementType(BpmnElementType.SERVICE_TASK)
+            .getFirst();
+
+    final Record<JobRecordValue> jobRecord =
+        RecordingExporter.jobRecords(expectedJobIntent)
+            .withProcessInstanceKey(processInstanceKey)
+            .skip(jobIndex)
+            .getFirst();
+
+    Assertions.assertThat(jobRecord.getValue())
+        .hasElementInstanceKey(activatingJob.getKey())
+        .hasElementId(activatingJob.getValue().getElementId())
+        .hasProcessDefinitionKey(activatingJob.getValue().getProcessDefinitionKey())
+        .hasBpmnProcessId(activatingJob.getValue().getBpmnProcessId())
+        .hasProcessDefinitionVersion(activatingJob.getValue().getVersion())
+        .hasJobKind(expectedJobKind)
+        .hasType(expectedJobType);
+  }
+
+  private void assertJobLifecycle(
```

Comment:
:wrench: The method name is misleading because it asserts and completes the job.

Reply:
Agreed, I've updated the method name to better reflect its functionality. 
Thanks for catching that!